edxval/api.py

```diff
@@ -316,27 +318,42 @@ def get_url_for_profile(edx_video_id, profile):
 
 def _get_videos_for_filter(
         video_filter,
-        sort_field=None,
-        sort_dir=SortDirection.asc
+        sort_field=VideoSortField.created,
+        sort_dir=SortDirection.desc,
+        **kwargs
 ):
     """
     Returns a generator expression that contains the videos found, sorted by
     the given field and direction, with ties broken by edx_video_id to ensure a
     total order.
     """
-    videos = Video.objects.filter(**video_filter)
+    videos_qs = Video.objects.filter(**video_filter)
     if sort_field:
         # Refining by edx_video_id ensures a total order
-        videos = videos.order_by(sort_field.value, "edx_video_id")
-        if sort_dir == SortDirection.desc:
-            videos = videos.reverse()
-    return (VideoSerializer(video).data for video in videos)
+        videos_qs = videos_qs.order_by(sort_field.value, "edx_video_id")
+
+    if sort_dir == SortDirection.desc:
+        videos_qs = videos_qs.reverse()
+
+    paginated = kwargs.pop("paginated", False)
+    if paginated:
+        page_no = int(kwargs.pop("page", 1))
+        page_size = int(kwargs.pop("page_size", settings.VAL_PAGE_SIZE))
```

Comment:
this function expects integer input will error out for invalid input. validation being handled from platform side to make sure valid query parameters are sent.